src/azure-cli/azure/cli/command_modules/container/custom.py

```diff
@@ -438,23 +413,33 @@ def _create_image_registry_credentials(registry_login_server, registry_username,
                                                               username=registry_username,
                                                               password=registry_password)]
     elif ACR_SERVER_DELIMITER in image.split("/")[0]:
-        if not registry_username:
-            try:
-                registry_username = prompt(msg='Image registry username: ')
-            except NoTTYException:
-                raise CLIError('Please specify --registry-username in order to use Azure Container Registry.')
-
-        if not registry_password:
-            try:
-                registry_password = prompt_pass(msg='Image registry password: ')
-            except NoTTYException:
-                raise CLIError('Please specify --registry-password in order to use Azure Container Registry.')
-
         acr_server = image.split("/")[0] if image.split("/") else None
-        if acr_server:
-            image_registry_credentials = [ImageRegistryCredential(server=acr_server,
-                                                                  username=registry_username,
-                                                                  password=registry_password)]
+        if identity:
+            if not is_valid_resource_id(identity):
+                msi_client = cf_msi(cmd.cli_ctx)
+                identity = msi_client.user_assigned_identities.get(resource_group_name=resource_group_name,
+                                                                   resource_name=identity).id
+            if acr_server:
+                image_registry_credentials = [ImageRegistryCredential(server=acr_server,
+                                                                      username=registry_username,
+                                                                      password=registry_password,
+                                                                      identity=identity)]
+        else:
+            if not registry_username:
+                try:
+                    registry_username = prompt(msg='Image registry username: ')
+                except NoTTYException:
+                    raise CLIError('Please specify --registry-username in order to use Azure Container Registry.')
+
+            if not registry_password:
+                try:
+                    registry_password = prompt_pass(msg='Image registry password: ')
+                except NoTTYException:
+                    raise CLIError('Please specify --registry-password in order to use Azure Container Registry.')
```

Comment:
Pease use the specific error type `RequiredArgumentMissingError` instead of `CLIError`

Reply:
Changed to more specific error type